x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/IndexResolver.java

```diff
@@ -151,13 +165,110 @@ public IndexResolution mergedMappings(String indexPattern, FieldCapabilitiesResp
             return IndexResolution.valid(new EsIndex(indexPattern, rootFields, Map.of()));
         }
 
+        Map<String, StringBuilder> clusterAndResolvedIndices = new HashMap<>();
         Map<String, IndexMode> concreteIndices = Maps.newMapWithExpectedSize(fieldCapsResponse.getIndexResponses().size());
         for (FieldCapabilitiesIndexResponse ir : fieldCapsResponse.getIndexResponses()) {
+            String indexExpression = ir.getIndexName();
             concreteIndices.put(ir.getIndexName(), ir.getIndexMode());
+
+            String clusterAlias = parseClusterAlias(indexExpression);
+            clusterAndResolvedIndices.compute(clusterAlias, (k, v) -> {
+                if (v == null) {
+                    return new StringBuilder().append(indexExpression);
+                } else {
+                    return v.append(',').append(indexExpression);
```

Comment:
Good point about it possibly being overly long and expensive. We don't need to, but since it was easy to add here, I tried to do it. But _search `_clusters/details` does not show the resolved indices - just the requested index expression from the user (so keeping the wildcards if any present), so I can remove this part and just keep the original index expression. That would make it consistent with _search.